src/user_version.rs

```diff
@@ -21,13 +28,19 @@ impl UserVersion {
             _ => None,
         }
     }
-}
 
-impl PartialEq<Version> for UserVersion {
-    fn eq(&self, other: &Version) -> bool {
-        match (self, other) {
+    pub fn matches(&self, version: &Version, config: &crate::config::FnmConfig) -> bool {
+        match (self, version) {
             (Self::Full(a), b) if a == b => true,
-            (Self::Full(_), _) => false,
+            (Self::Full(user_version), maybe_alias) => {
+                match (user_version.alias_name(), maybe_alias.find_aliases(&config)) {
```

Comment:
I wonder that this wouldn't be as fast as we can, but I don't try to optimize that right now.